pkg/ec2metadata/ec2metadata.go

```diff
@@ -196,16 +196,17 @@ func (e *Service) GetRebalanceRecommendationEvent() (rebalanceRec *RebalanceReco
 // GetMetadataInfo generic function for retrieving ec2 metadata
 func (e *Service) GetMetadataInfo(path string) (info string, err error) {
 	resp, err := e.Request(path)
-	if resp != nil && (resp.StatusCode < 200 || resp.StatusCode >= 300) {
-		return "", fmt.Errorf("Metadata request received http status code: %d", resp.StatusCode)
-	}
-	if err != nil {
+	if err != nil || resp == nil {
 		return "", fmt.Errorf("Unable to parse metadata response: %w", err)
```

Comment:
nit: In the event of `resp == nil` and `err == nil` both being `true`, this log would look strange. Should we have separate log for "response was empty"?

Reply:
Based on the three `return` statements in `Request()`, it's not possible for them to both be nil (i.e., if `resp` is nil, then `err` will describe why). So it's sufficient to check only `err != nil`.

I'll remove the `|| resp == nil` part of the statement to clarify that and keep it consistent with the style of the rest of this function.

Reply:
Based on the three `return` statements in `Request()`, it's not possible for them to both be nil (i.e., if `resp` is nil, then `err` will describe why). So it's sufficient to check only `err != nil`.

I'll remove the `|| resp == nil` part of the statement to clarify that and keep it consistent with the style of the rest of this function.

Reply:
Discussed offline. We did not previously have a nil check on `resp`, which is probably incorrect. I've added one in the most recent commit.